src/main/webapp/dev/js/main/instructorHelp.js

```diff
@@ -25,7 +34,224 @@ function bindPanelCollapseLinksAndAnchor() {
     });
 }
 
+function prepareQuestionsForSearch() {
+    let questionCount = 0;
+
+    function addQuestionsForTopic(topic) {
+        const topicContentElement = $(topic).next();
+
+        topicContentElement.find('.panel-group').each((idx, subTopic) => {
+            const subTopicText = $(subTopic).prev().text();
+            const subTopicQuestions = $(subTopic).children();
+
+            let subTopicTags = subTopicText.split(' ').map((word) => {
+                // ensure case insensitivity
+                const lowerCaseWord = word.toLowerCase();
+
+                // ignore 1 and 2 letter words which are probably articles and prepositions
+                if (lowerCaseWord.length < 3) {
+                    return '';
+                }
+                return elasticlunr.stemmer(lowerCaseWord);
```

Comment:
Be careful with stemming; while it's useful for searching, if the search matches only the stemmed version but not the original version, there will be no visual evidence of the match through highlighting so results might seem inconsistent.

e.g. search for `teams` -- search (correctly) retrieves questions that deal with `team`, but since only exact matches are highlighted, some questions are deemed relevant without displaying any highlights at all. 

For consistency, I would prefer if either we highlighted words whose stemmed forms match (e.g. highlight `team` even though search term was `teams` and vice versa), or don't highlight any words.

Reply:
I agree with your point. I was experimenting with this yesterday and basically if I remove stemming altogether from the processing then only those results that contain the exact match are returned in majority of cases. This solves the highlight problem but we miss out on potentially relevant questions whose stems match.

If we highlight the stems instead we might get wierd cases due to the way stemmer works. For example, the stem for "course" and "courses" is "cours" which might not make sense if highlighted ?

I think we should keep highlighting since it helps in picking relevant information in large panels.